azavea/pfb-network-connectivity
# Feature/kak/store analysis results#587

## Overview

Supersedes #599.

Adds two models for the two exported shapefiles, to store each geometry in the shapefile as a separate object, along with its properties necessary for map tiling.


### Demo

Inspecting created objects at the console:
```
>>> NeighborhoodWaysResults.objects.first().geom.json
'{"type": "LineString", "coordinates": [[-106.336278600001, 42.8248898000048], [-106.335824300001, 42.8249606000048], [-106.335458400001, 42.8250697000048], [-106.335032200001, 42.8252319000048], [-106.334606, 42.8254471000048]]}'
>>> CensusBlocksResults.objects.first().geom.json
'{"type": "MultiPolygon", "coordinates": [[[[-106.35391, 42.7742500000048], [-106.35367, 42.7746600000048], [-106.35348, 42.7749700000048], [-106.35334, 42.7751500000048], [-106.35306, 42.7750300000048], [-106.35305, 42.7749500000048], [-106.35318, 42.7747900000048], [-106.35328, 42.7746100000048], [-106.35336, 42.7744800000048], [-106.35336, 42.7743700000048], [-106.35341, 42.7743], [-106.35352, 42.7741900000048], [-106.35359, 42.7740600000048], [-106.35362, 42.7739500000048], [-106.35362, 42.7737900000048], [-106.35379, 42.7738], [-106.3541, 42.7738800000048], [-106.354082, 42.7739130000048], [-106.35391, 42.7742500000048]]]]}'
>>> CensusBlocksResults.objects.first().job      
<AnalysisJob: <AnalysisJob: COMPLETE Casper>>
>>> NeighborhoodWaysResults.objects.first().job      
<AnalysisJob: <AnalysisJob: COMPLETE Casper>>
```


### Notes

`LayerMapping` ([source](https://github.com/django/django/blob/master/django/contrib/gis/utils/layermapping.py)) does not provide a means for setting model fields that do not come from the shapefile, or to get at the objects created from the shapefile, but we need to set the related job on the objects to be able to identify them later. To deal with this, the data migration assumes any created objects without a job set already came from the last imported shapefile, and set the job to the current job. This will work for the migration, as we can expect it to run serially and for no other operations to happen on the newly added models at the same time, but we will need to find some other way to set the `job` in whatever later task handles adding these objects at the time the shapefile-creating analysis job is run. The easiest solution for future analysis jobs may be to simply add a field to the exported shapefile for the job UUID, which could then be used as part of the field mapping.


## Testing Instructions

 * Run migrations
 * Go to Django console
 * Expected properties should exist on the created objects
 * All fields used by the current tiler stylings should exist on the new models


Closes #587
